.github/renovate.json5

```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:recommended",
+    ":semanticCommitsDisabled",
+    "customManagers:githubActionsVersions"
+  ],
+  "labels": ["dependencies"],
+  "postUpdateOptions": ["gomodTidy"],
+  "automergeStrategy": "squash",
+  // required for automerging patch updates
+  "separateMinorPatch": true,
+  "customManagers": [
+    {
+      // Update `_VERSION` and `_version` variables in Makefiles and scripts.
+      // Inspired by `regexManagers:dockerfileVersions` preset.
+      "customType": "regex",
+      "fileMatch": ["Makefile$", "\\.mk$", "\\.sh$"],
+      "matchStrings": [
+        "# renovate: datasource=(?<datasource>[a-z-.]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[^\\s]+?))?(?: extractVersion=(?<extractVersion>[^\\s]+?))?(?: registryUrl=(?<registryUrl>[^\\s]+?))?\\s.+?_(VERSION|version) *[?:]?= *\"?(?<currentValue>.+?)\"?\\s"
+      ]
+    }
+  ],
+  "packageRules": [
+    {
+      // automerge patch updates
+      "matchUpdateTypes": ["patch"],
+      "automerge": true
+    },
+    {
+      // bump k8s, controller-runtime and gardener/gardener go dependencies together
+      "groupName": "k8s packages",
+      "groupSlug": "k8s-go",
+      "matchDatasources": ["go"],
+      "matchPackagePrefixes": [
+        // from "group:kubernetes"
+        "k8s.io/api",
+        "k8s.io/apiextensions-apiserver",
+        "k8s.io/apimachinery",
+        "k8s.io/apiserver",
+        "k8s.io/cli-runtime",
+        "k8s.io/client-go",
+        "k8s.io/cloud-provider",
+        "k8s.io/cluster-bootstrap",
+        "k8s.io/code-generator",
+        "k8s.io/component-base",
+        "k8s.io/controller-manager",
+        "k8s.io/cri-api",
+        "k8s.io/csi-translation-lib",
+        "k8s.io/kube-aggregator",
+        "k8s.io/kube-controller-manager",
+        "k8s.io/kube-proxy",
+        "k8s.io/kube-scheduler",
+        "k8s.io/kubectl",
+        "k8s.io/kubelet",
+        "k8s.io/legacy-cloud-providers",
+        "k8s.io/metrics",
+        "k8s.io/mount-utils",
+        "k8s.io/pod-security-admission",
+        "k8s.io/sample-apiserver",
+        "k8s.io/sample-cli-plugin",
+        "k8s.io/sample-controller",
+        // added packages
+        "sigs.k8s.io/controller-runtime",
+        "github.com/gardener/gardener"
+      ]
+    },
+    {
+      // automerge k8s.io/utils updates
+      "matchDatasources": ["go"],
+      "matchPackageNames": ["k8s.io/utils"],
+      "matchUpdateTypes": ["digest"],
```

Comment:
Why is this needed and how does this work?

Reply:
I'm not sure if I get your question. The package is always referenced by digest and this is how you tell renovate to automerge updates: https://github.com/stackitcloud/gardener-extension-acl/blob/981945f35dc19535e3d4e97ed9311c326b8ca75b/go.mod#L31

Reply:
I was wondering, how does renovate know if a package is newer, if it uses a digest instead of semver. But based on this example it could be alphabetical.